src/graph.py

```diff
@@ -11,9 +11,11 @@ def __init__(self, initial_graph=None):
          3:[1]}
         '''
         if initial_graph is None:
-            self.graph = {}
-        else:
+                self.graph = {}
+        elif isinstance(initial_graph, dict):
             self.graph = initial_graph
+        else:
+            raise TypeError("Please pass a dictionary")
 
     def __iter__(self):
         return iter(self.graph)
```

Comment:
Your current `edges()` implementation isn't quite up to snuff; it will return a list of lists of nodes. This won't mean much as edges since you wouldn't know what is the starting node for each edge.

To represent edges, the method could return e.g. a list of tuples of the from node and the to node i.e. `[('a', 'b'), ('a', 'c'), ('b', 'd')]` etc. or somethng like `['a -> b', 'a -> c', 'b -> d']`.